app/controllers/hyrax/file_sets_controller.rb

```diff
@@ -137,11 +151,24 @@ def search_builder_class
 
       def initialize_edit_form
         @parent = @file_set.in_objects.first
+        workflow_check(parent_id: @parent.id)
         original = @file_set.original_file
         @version_list = Hyrax::VersionListPresenter.new(original ? original.versions.all : [])
         @groups = current_user.groups
       end
 
+      def workflow_check(parent_id: nil)
+        return if current_ability.can?(:edit, @solr_document)
+        if parent_id.nil?
+          curation_concern = ::FileSet.find(params[:id])
+          parent = curation_concern.parent
+          parent_id = parent.id
+        end
+        doc = ::SolrDocument.find(parent_id)
+        return if current_ability.can?(:edit, doc)
+        raise WorkflowAuthorizationException if doc.suppressed? && current_ability.can?(:read, doc)
```

Comment:
i'm not following this line. do we need to check `current_ability.can?(:read, doc)`?

what happens if ` doc.suppressed?` is `true` but the current ability *cannot* read the doc?

Reply:
I copied this code from elsewhere. And yes, it looks like it is redundant and/or misleading. I've removed the trailing `current_ability.can?(:read, doc)`